core_backend/app/languages/routers.py

```diff
@@ -0,0 +1,182 @@
+from typing import Annotated, List
+
+from fastapi import APIRouter, Depends
+from fastapi.exceptions import HTTPException
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from ..auth.dependencies import get_current_fullaccess_user, get_current_readonly_user
+from ..auth.schemas import AuthenticatedUser
+from ..database import get_async_session
+from ..utils import setup_logger
+from .models import (
+    LanguageDB,
+    delete_language_from_db,
+    get_default_language_from_db,
+    get_language_from_db,
+    get_list_of_languages_from_db,
+    is_language_name_unique,
+    save_language_to_db,
+    update_language_in_db,
+)
+from .schemas import LanguageBase, LanguageRetrieve
+
+router = APIRouter(prefix="/languages")
+logger = setup_logger()
+
+
+@router.post("/", response_model=LanguageRetrieve)
+async def create_language(
+    language: LanguageBase,
+    full_access_user: Annotated[
+        AuthenticatedUser, Depends(get_current_fullaccess_user)
+    ],
+    asession: AsyncSession = Depends(get_async_session),
+) -> LanguageRetrieve | None:
+    """
+    Create language endpoint. Upsert language to PG database
+    """
+    language.language_name = language.language_name.upper()
+    if not (await is_language_name_unique(language.language_name, asession)):
+        raise HTTPException(status_code=400, detail="Language name already exists")
+
+    if language.is_default is True:
+        await unset_default_language(asession)
+
```

Comment:
Maybe we should have another check that checks if there is a default language and if not, returns an error and says "No default language found. Please either create a default language or set an existing language as default."? Not sure about this pattern though. Any other ideas for how to ensure there is always a default language?

Reply:
Right now  I can just make sure we cannot update the default language to non_default. Telling the user to directly set the new default language.